extensions/wikia/AdEngine/js/template/bfaa.js

```diff
@@ -124,14 +130,20 @@ define('ext.wikia.adEngine.template.bfaa', [
 				return log(['show', 'not supported skin'], 'info', logGroup);
 		}
 
+		log(['show', page, wrapper, params], 'info', logGroup);
+
 		wrapper.style.opacity = '0';
 		slotTweaker.makeResponsive(params.slotName, params.aspectRatio);
 		slotTweaker.onReady(params.slotName, function (iframe) {
 			handler.show(iframe, params.aspectRatio);
 			wrapper.style.opacity = '';
+
+			if (recoveryHelper.isBlocking()) {
+				slotTweaker.tweakRecoveredSlot(iframe.parentNode.parentNode.parentNode.parentNode);
```

Comment:
this one doesn't look very robust. Is there a way to make it more dynamic? like get first parent matching particular selector?